notebooks/linear_regression.livemd

```diff
@@ -0,0 +1,610 @@
+  scale: [domain: [-124.55, -113.80]]
+)
+|> Vl.encode_field(:y, "latitude",
+  type: :quantitative,
+  axis: [grid: false],
+  scale: [domain: [32.45, 42.05]]
+)
+```
+
+We can see better the more dense areas of the plot. Now we will focus more on house prices. Encode them in color on the map. The size of a circle will indicate the population of districts.
+
+```elixir
+Vl.new(
+  title: [
+    text: "Distribution of houses across California",
+    offset: 20
+  ],
+  width: 630,
+  height: 630
+)
+|> Vl.data_from_values(df)
+|> Vl.mark(:circle, opacity: 0.4)
+|> Vl.encode_field(:x, "longitude",
+  type: :quantitative,
+  axis: [grid: false],
+  scale: [domain: [-124.55, -113.80]]
+)
+|> Vl.encode_field(:y, "latitude",
+  type: :quantitative,
+  axis: [grid: false],
+  scale: [domain: [32.45, 42.05]]
+)
+|> Vl.encode_field(:color, "median_house_value", type: :quantitative, scale: [scheme: :viridis])
+|> Vl.encode_field(:size, "population", type: :quantitative)
+```
+
+In the end, we may also add the map of California.
+
+```elixir
+Vl.new(
+  title: [
+    text: "Scatterplot of Generated Data",
+    offset: 20
+  ],
+  width: 630,
+  height: 630
+)
+|> Vl.layers([
+  Vl.new()
+  |> Vl.data_from_values([
+    %{
+      url:
+        "https://raw.githubusercontent.com/ageron/handson-ml2/master/images/end_to_end_project/california.png"
+    }
+  ])
+  |> Vl.mark(:image,
+    width: 630,
+    aspect: false,
+    align: :right
+  )
+  |> Vl.encode_field(:url, "url", type: :nominal),
+  Vl.new()
+  |> Vl.data_from_values(df)
+  |> Vl.mark(:circle, opacity: 0.4)
+  |> Vl.encode_field(:x, "longitude",
+    type: :quantitative,
+    axis: [grid: false],
+    scale: [domain: [-124.55, -113.80]]
+  )
+  |> Vl.encode_field(:y, "latitude",
+    type: :quantitative,
+    axis: [grid: false],
+    scale: [domain: [32.45, 42.05]]
+  )
+  |> Vl.encode_field(:color, "median_house_value", type: :quantitative, scale: [scheme: :viridis])
+  |> Vl.encode_field(:size, "population", type: :quantitative)
+])
+```
+
+From This plot, we can read that prices are substantially dependent on **geolocalization** and **population**. For geolocalization, we see, those areas closer to the ocean are more expensive. But it's not a strict rule since houses on the northern bay of California are much more affordable than in in-land Mid California. For the population, there are two dense areas with expensive housing: Los Angeles Bay (In South California) and San Francisco Bay (in Mid Califonia). They are metropolises with a lot of different tech companies, and business and cultural institutions, so, logically, housing in those places will be expensive.
+
+<ins>
+<i>
+Hint:
+</i>
+</ins>
+
+<br />
+
+You can try to add another feature by computing clustering on this data set. It might be a sum or power mean of distances to the clusters. We may predict that centroids will be located in San Francisco Bay and Los Angeles Bay. You can also pass population as weights to k-means.
```

Comment:
```suggestion
You can try to add another feature by computing clustering on this data set. It might be a sum or power mean of distances to the clusters. We may predict that centroids will be located in San Francisco Bay and Los Angeles Bay. You can also pass population as weights to K-means.
```